### PR TITLE
Feat: Queries Gen 2

### PR DESCRIPTION
**Pull request by Meldiron**
Implementation of new queries syntax.

Goal of new implementation: https://github.com/appwrite/appwrite/discussions

- [x] Existing tests updated
- [x] Implemented new tests

### CODE REVIEW

File: src/Database/Query.php [hunk 1]
```diff
@@ -140,106 +105,130 @@ public static function isOperator(string $value): bool
...
+            $param = substr($param, 1, -1); // Remove '' or ""
+            return $param;
+        }
+
+        // Unknown format
+        return $param;
     }
 }
+
+Query::$TYPE_ALIASES = [
```
Comment (eldadfux) [inline src/Database/Query.php:227]: This is an anti-pattern. + we can and should treat aliases similar to how we use regular methods to help reduce complexity of this class.

File: src/Database/Validator/QueryValidator.php [hunk 2]
```diff
@@ -90,32 +90,32 @@ public function getDescription(): string
     public function isValid($query): bool
     {
         // Validate operator
-        if (!in_array($query->getOperator(), $this->operators)) {
-            $this->message = 'Query operator invalid: ' . $query->getOperator();
+        if (!in_array($query->getMethod(), $this->operators)) {
```
Comment (eldadfux) [inline src/Database/Validator/QueryValidator.php:93]: If we move from operators to methods we should adapt the terminology library-wide.

**Review by eldadfux — CHANGES_REQUESTED** [review]
Some thoughts:
1. We should try and make the Query class behave closer to our external API for consistency, we can have the same existing constructor with some factory method to represent different methods. 
2. I feel the Query interface can tell a better story. A lot of the new methods could be private, and expose a ,ore intuitive interface, I would love to rethink the structure together.

File: tests/Database/Validator/QueriesTest.php [hunk 3]
```diff
@@ -104,8 +104,8 @@ public function setUp(): void
 
         $this->queryValidator = new QueryValidator($attributes);
 
-        $query1 = Query::parse('title.notEqual("Iron Man", "Ant Man")');
-        $query2 = Query::parse('description.equal("Best movie ever")');
+        $query1 = Query::parse('notEqual("title", ["Iron Man", "Ant Man"])');
+        $query2 = Query::parse('equal("description", "Best movie ever")');
```
Comment (stnguyen90) [inline tests/Database/Validator/QueriesTest.php:108]: @Meldiron, would you please make sure to have a test for text with a comma to address https://github.com/appwrite/appwrite/issues/3089? Foe example:

```php
 $query = Query::parse('equal("text", ["Contains,Comma", "No Comma"])');
 ```

**Review by stnguyen90 — CHANGES_REQUESTED** [review]

**Comment by Meldiron** [comment]
Closing in favour of new branch `2.1`

File: src/Database/Query.php (hunk 1 above)
Comment (Meldiron) [reply src/Database/Query.php:227]: We simplified aliases. Tho we can't support the page now, the system is much simpler. We can allow complex aliases in future iterations.

File: tests/Database/Validator/QueriesTest.php (hunk 3 above)
Comment (Meldiron) [reply tests/Database/Validator/QueriesTest.php:108]: Addressed.

File: src/Database/Validator/QueryValidator.php (hunk 2 above)
Comment (Meldiron) [reply src/Database/Validator/QueryValidator.php:93]: Adapted. All operators renamed to methods.

**Pull request closed without merging**